src/widgets/newsFeed/api/feedApi.tsx

```diff
@@ -0,0 +1,40 @@
+import { useInfiniteQuery } from "react-query";
+
+
+type useInfiniteArticlesProps = {
+  queryKey: unknown[],
+  queryFn: unknown,
+  query: unknown,
+  params?: unknown
+}
+
+
+const useInfiniteArticles = ({
+                               queryKey,
+                               queryFn,
+                               query,
+                               params
+                             }: useInfiniteArticlesProps) => {
+  const { offset, limit } = query;
+
+  return useInfiniteQuery<Article[]>({
+    queryKey, queryFn: async ({ pageParam = offset }) => {
+      const response = await queryFn(
+          { ...query, offset: pageParam },
+          { ...params }
+      );
+
+      return response.data.articles.map(mapArticle);
+    }
+  })
+
+}
+
+
+export const useInfiniteArticlesFeed = {
+  queryKey: 'ArticlesFeed',
+  queryFn: async () => {
+    const responce = await fetch('http://localhost:8000/articles')
+    return await responce.json()
+  }
+}
```

Comment:
need to refactor it
1. All requests should be somewhere in api
2. Work with data in entities
3. Infinity scroll should include limit (default number of items)